src/ArraySetops.chpl

```diff
@@ -39,6 +42,31 @@ module ArraySetops
 
       return boolIndexer(head, mask);
     }
+
+    proc intersect1d_multi(segments1: borrowed SymEntry(int), values1: borrowed SymEntry(?t), lens1: [] int, segments2: borrowed SymEntry(int), values2: borrowed SymEntry(t), lens2: [] int, isUnique: bool) throws {
+      var intx_lens: [segments1.aD] int;
+
+      // Compute lengths of the segments resulting from each union
+      forall (i, s1, l1, s2, l2, il) in zip(segments1.aD, segments1.a, lens1, segments2.a, lens2, intx_lens) with (var agg = newDstAggregator(int)){
+        // TODO - update to use lowLevelLocalizingSlice 
+        var intx = intersect1d(values1.a[s1..#l1], values2.a[s2..#l2], isUnique);
+        agg.copy(il, intx.size);
+      }
+
+      const intx_segs = (+ scan intx_lens) - intx_lens;
+      var intx_vals = makeDistArray((+ reduce intx_lens), t);
+
+      // Compute the union and add values to the corresponding indexes in values
+      forall (i, s1, l1, s2, l2, is, il) in zip(segments1.aD, segments1.a, lens1, segments2.a, lens2, intx_segs, intx_lens) with (var agg = newDstAggregator(t)){
+        // TODO - update to use lowLevelLocalizingSlice 
+        var intx = intersect1d(values1.a[s1..#l1], values2.a[s2..#l2], isUnique);
+        for i in (0..#il){
+          agg.copy(intx_vals[i+is], intx[i]);
+        }
```

Comment:
Thanks for catching this. I didn't even think about it. I made this change in each setop function.

Reply:
I don't fully understand what this code is trying to do, but at a glance `values1.a[s1..#l1]` should be creating a local array, and I don't think `intersect1d` would do anything to then distributed a local array, but I haven't checked. So in the current code I _think_ `intx` should always be local.

That said, I don't think this code will be fast because it will take time to create those local arrays, when in the common case you're probably just operating on a local portion of the distributed array. This is was `lowLevelLocalizingSlice` was originally added for -- to make it cheaper to operate on portions of likely local, but possibly remote arrays.

However, for sorted strings Bill was seeing performance for bringing the data to the current node was quite poor because it was no longer the common case that data was local, so he introduced the `computeOnSegments` code as a way to always operate on local portions of strings. https://github.com/Bears-R-Us/arkouda/issues/1101 has more information on that, and I wonder if extending `computeOnSegments` is what you'd want to be doing here.

Reply:
I believe you may be correct for intersect. The other functions are all using the uniqueSort, which returns a new distributed array. I just lumped everything together. I can update that.

This code is definitely not going to be the best performance wise. The intention was to move these operations to the server because it will still perform better than handling everything client side. I have not yet, but will be adding an issue to address the performance of these set operations.

Reply:
I agree with Ethan, intersect is calling uniquesort which is returning a distributed array

Reply:
Hmm, I thought most of the sort routines had the output type match the input type, but I could easily be wrong here (and I haven't looked at the code yet, so I trust what you're saying over my guess)